crates/noirc_frontend/src/graph/mod.rs

```diff
@@ -83,7 +83,11 @@ impl CrateGraph {
     pub fn add_crate_root(&mut self, crate_type: CrateType, file_id: FileId) -> CrateId {
         let mut roots_with_file_id =
             self.arena.iter().filter(|(_, crate_data)| crate_data.root_file_id == file_id);
-        assert!(roots_with_file_id.next().is_none(), "you cannot add the same file id twice");
```

Comment:
So were we hitting this assert?

Reply:
yep, it must have hit it before as it was adding the same path twice